src/components/Molecules/AddressOverview/AddressOverview.tsx

```diff
@@ -0,0 +1,297 @@
+import { TokenAvatar } from "@/components/Atoms/TokenAvatar/TokenAvatar";
+import { Button } from "@/components/ui/button";
+import {
+    Card,
+    CardContent,
+    CardDescription,
+    CardTitle,
+} from "@/components/ui/card";
+import { Skeleton } from "@/components/ui/skeleton";
+import { GRK_SIZES } from "@/utils/constants/shared.constants";
+import { None, Some, type Option } from "@/utils/option";
+import { useCovalent } from "@/utils/store/Covalent";
+import { type AccountOverviewProps } from "@/utils/types/molecules.types";
+import {
+    calculatePrettyBalance,
+    type BalanceItem,
+    type Chain,
+    prettifyCurrency,
+} from "@covalenthq/client-sdk";
+import {
+    DropdownMenu,
+    DropdownMenuContent,
+    DropdownMenuItem,
+    DropdownMenuSeparator,
+    DropdownMenuTrigger,
+} from "@radix-ui/react-dropdown-menu";
+import { CaretDownIcon } from "@radix-ui/react-icons";
+import { useCallback, useEffect, useState } from "react";
+
+export const AddressOverview: React.FC<AccountOverviewProps> = ({
+    address,
+    chain_names,
+}) => {
+    const { covalentClient, chains } = useCovalent();
+
+    const [maybeResult, setResult] = useState<
+        Option<
+            {
+                chain_name: Chain;
+                balances: BalanceItem[];
+                native: BalanceItem;
+            }[]
+        >
+    >(None);
+
+    useEffect(() => {
+        (async () => {
+            setResult(None);
+            const results: {
+                chain_name: Chain;
+                balances: BalanceItem[];
+                native: BalanceItem;
+            }[] = [];
+            await Promise.all(
+                chain_names.map(async (chain_name) => {
```

Comment:
Lets change this to single chain for now, this component wont scale well if multiple chains are selected. The UI will look a bit off if they display every chain

Reply:
Fixed in: 22f5672